internetarchive/files.py

```diff
@@ -272,8 +273,10 @@ def download(self, file_path=None, verbose=None, ignore_existing=None,
         except (RetryError, HTTPError, ConnectTimeout, OSError, ReadTimeout) as exc:
             msg = f'error downloading file {file_path}, exception raised: {exc}'
             log.error(msg)
-            if os.path.exists(file_path):
-                os.remove(file_path)
+            try:
+                os.remove(file_path.encode('utf-8'))
```

Comment:
I cannot produce a case where explicitly encoding as UTF-8 would be necessary or sensible. It would probably even be harmful on systems which do not use UTF-8.

Reply:
There are a bunch of other file path utf8 encodes in this file, I added this one because I did run into a file downloaded from the archive where this line threw an exception without the encoding. So I vote either leave it, or remove all the other ones too.

Reply:
Could you please provide a link to the item and file that produced the error?

I tested it with various characters and could not cause an exception.
If it is really necessary then we should change it to encode to `sys.getfilesystemencoding()` everywhere and even add it in the places where it is currently missing.

Reply:
I was wrong on this one, it just took a while to resurface because I'm working with such a large dataset. I made an assumption because it was a weird unicode filename and there were other utf8 encodes in the same file, but the filename was actually just way too long, and unicode was masking that.

My sys.getfilesystemencoding() is actually utf8 already, which means the encoding is doing nothing. Mild vote to remove the other encodes.